pmm-app/src/pmm-settings/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {
+  FC,
+  useState
+} from 'react';
+import { Form as FinalForm, Field as FinalField, FormRenderProps } from 'react-final-form';
+import {
+  Button,
+  LinkButton,
+  useTheme,
+} from '@grafana/ui';
+import { showSuccessNotification, showErrorNotification } from 'shared/components/helpers';
+import { ButtonWithSpinner } from 'shared/components/Form';
+import validators from 'shared/components/helpers/validators';
+import { getStyles } from './SignUp.styles';
+import { Messages } from './SignUp.messages';
+import { InputFieldAdapter, CheckboxFieldAdapter } from './FieldAdapters/FieldAdapters';
+import { TERMS_OF_SERVICE_URL, PRIVACY_POLICY_URL, NOTIFICATION_SETTINGS_URL } from './SignUp.constants';
+import { SignUpData, SignUpProps } from './types';
+import { LoggedIn } from './LoggedIn/LoggedIn';
+
+const signUp = async (signUpData: SignUpData) => {
+  // XXX: stub for the actual sign up API call
+  signUpData;
+  await new Promise((resolve) => setTimeout(resolve, 1500));
+};
+
+export const SignUp: FC<SignUpProps> = ({ userEmail }) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+  const [loggedInEmail, setLoggedInEmail] = useState(userEmail);
+
+  const CheckboxLabel = () => (
+    <span data-qa="sign-up-agreement-checkbox-label" className={styles.checkboxLabel}>
+      {Messages.agreementFirstPart}
+      {' '}
+      <LinkButton className={styles.link} variant="link" href={TERMS_OF_SERVICE_URL}>{Messages.termsOfService}</LinkButton>
+      ,
+      {' '}
+      <LinkButton className={styles.link} variant="link" href={PRIVACY_POLICY_URL}>{Messages.privacyPolicy}</LinkButton>
+      ,
+      {' '}
+      {Messages.agreementSecondPart}
+      {' '}
+      <LinkButton className={styles.link} variant="link" href={NOTIFICATION_SETTINGS_URL}>{Messages.notificationSettings}</LinkButton>
+    </span>
+  );
+
+  const handleSignUpFormSubmit = async (signUpData: SignUpData) => {
+    try {
+      await signUp(signUpData);
+      setLoggedInEmail(signUpData.email);
+      showSuccessNotification({ message: Messages.signUpSucceeded });
+    } catch (e) {
+      console.error(e);
+      showErrorNotification({ message: Messages.errors.signUpFailed });
+    }
+  };
+
+  const InnerForm: FC<FormRenderProps<SignUpData>> = ({
+    pristine, submitting, valid, handleSubmit
+  }) => (
+    <form data-qa="sign-up-form" className={styles.form} onSubmit={handleSubmit}>
+      <legend className={styles.legend}>
+        {Messages.signUp}
+      </legend>
+      <FinalField
+        data-qa="sign-up-email-input"
+        name="email"
+        label="Email"
```

Comment:
Should these labels be in the messages file?

Reply:
Good point, done